src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,14 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * Returns whether the command type will mutate data
+     *
+     */
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implemented in child classes");
```

Comment:
Probably better to implement as an abstract class instead of throwing an exception.